src/smtp_server.c

```diff
@@ -235,7 +249,6 @@ int __smtp_server_run(struct smtp_server_context *ctx, bfd_t *stream)
 			n++;
 		}
 
-		/* Invoke all command handlers */
 		continue_session = smtp_server_process(ctx, c, c + n, stream);
```

Comment:
'continue_session' is not used anywhere else. I suggest removing variable and calling smtp_server_process() directly inside while() on the next line.